### ISSUE TITLE
java.lang.RuntimeException: Table clientes does not contain a primary key and can not be anonymyzed

### ISSUE THREAD

**Issue by PedroAvelar** [realrolfje/anonimatron#113]
Hello my friends, 

I am trying to run Anonimatron on my Oracle table to perform a test and I am getting the error described in the title.

The problem is that my table has a primary key, check it out:

```
-- Create table
create table CLIENTES
(
  id INTEGER PRIMARY KEY,
  nome             VARCHAR2(200),
  email            VARCHAR2(150),
  cep              VARCHAR2(50),
  rg               VARCHAR2(50),
  cnpj             VARCHAR2(70),
  local_nascimento VARCHAR2(70),
  estado_civil     VARCHAR2(30),
  numero_endereco  INTEGER
);
```

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration jdbcurl="jdbc:oracle:thin:@my_host:my_port:my_sid"
               userid="my_user" password="my_strong_pass">
 
  <table name="clientes">
    <column name="nome" type="ROMAN_NAME" />
  </table>
  
</configuration>
```

OBS: 
I put fictitious information in the xml

**Comment by realrolfje** [comment]
Anonimatron gets the primary keys from the metadata information in the database, and checks to see if these columsn are not part of the configuration. Is the "nome" column the only column that needs to be anonymized in your original configuration?

**Comment by realrolfje** [comment]
You also check if your table has primary keys with the following query:
```sql
SELECT cols.table_name, cols.column_name, cols.position, cons.status, cons.owner
FROM all_constraints cons, all_cons_columns cols
WHERE cols.table_name = 'CLIENTES'
AND cons.constraint_type = 'P'
AND cons.constraint_name = cols.constraint_name
AND cons.owner = cols.owner
ORDER BY cols.table_name, cols.position;
```

**Comment by PedroAvelar** [comment]
Hello @realrolfje , tks for reply!

In fact, the "clientes" table is just a test that I am doing with Anonimatron.

The intention is to use it in all of our Oracle databases and in hundreds of tables.

The "name" field is not the primary key of the table, but the ID field.

Anonimatron worked perfectly on our Mysql and PostgreSQL databases but we have this problem with Oracles and SQL Servers.

Here is the result of the query you requested:

`SQL> SELECT cols.table_name, cols.column_name, cols.position, cons.status, cons.owner
  2  FROM all_constraints cons, all_cons_columns cols
  3  WHERE cols.table_name = 'CLIENTES'
  4  AND cons.constraint_type = 'P'
  5  AND cons.constraint_name = cols.constraint_name
  6  AND cons.owner = cols.owner
  7  ORDER BY cols.table_name, cols.position;
  
  

TABLE_NAME                     COLUMN_NAME                                                                        POSITION STATUS   OWNER
------------------------------ -------------------------------------------------------------------------------- ---------- -------- ------------------------------
CLIENTES                       ID                                                                                        1 ENABLED  PE_AVELAR

SQL> `

**Comment by PedroAvelar** [comment]
![image](https://user-images.githubusercontent.com/15635801/97201599-62c97a00-1791-11eb-83d8-8affb14e7ceb.png)

**Comment by PedroAvelar** [comment]
![image](https://user-images.githubusercontent.com/15635801/97201744-91dfeb80-1791-11eb-82e6-c356de0df669.png)

**Comment by realrolfje** [comment]
Thanks for the information. There was a very recent fix for issue #105 which could be related, but that would mean that you have a "clientes" table in two databases which are accessible by the same user.

I will try to create an environment matching yours to see if I can reproduce the problem locally.

**Comment by realrolfje** [comment]
Pedro, which version of Oracle database are you running?

**Comment by PedroAvelar** [comment]
Hi @realrolfje,

I managed to solve this problem.

By default, Anonimatron only masks tables that have a primary key, but the error message I posted here is generic and does not report the nature of the problem.
For Anonimatron works in Oracle databases, it is necessary that the names of the tables and columns in the XML file are written in capital letters.
In doing so, it worked perfectly.

Answering your last question, we have instances in versions 11, 12 and 19.

Thanks for your attention and all the help.
